c7n/resources/timestream.py

```diff
@@ -0,0 +1,137 @@
+from c7n.manager import resources
+from c7n.actions import Action
+from c7n.query import DescribeSource, QueryResourceManager, TypeInfo
+from c7n.utils import local_session, type_schema
+from c7n.tags import (
+    TagDelayedAction,
+    TagActionFilter,
+    Tag as TagAction,
+    RemoveTag as RemoveTagAction
+)
+
+
+class DescribeTimestream(DescribeSource):
+    def augment(self, resources):
+        for r in resources:
+            client = local_session(self.manager.session_factory).client('timestream-write')
+            r['Tags'] = client.list_tags_for_resource(ResourceARN=r['Arn'])['Tags']
+        return resources
+
+
+@resources.register('timestream-database')
+class TimestreamDatabase(QueryResourceManager):
+    class resource_type(TypeInfo):
+        service = 'timestream-write'
+        arn_type = ''
+        name = 'DatabaseName'
+        id = arn = 'Arn'
+        enum_spec = ('list_databases', 'Databases', {})
+        permission_prefix = 'timestream'
+        permissions = ('timestream:ListDatabases', )
+
+    source_mapping = {
+        'describe': DescribeTimestream,
+    }
+
+
+@resources.register('timestream-table')
+class TimestreamTable(QueryResourceManager):
+    class resource_type(TypeInfo):
+        service = 'timestream-write'
+        arn_type = ''
+        name = 'TableName'
+        id = arn = 'Arn'
+        enum_spec = ('list_tables', 'Tables', {})
+        permission_prefix = 'timestream'
+        permissions = ('timestream:ListTables', )
+
+    source_mapping = {
+        'describe': DescribeTimestream,
+    }
+
+
+@TimestreamDatabase.action_registry.register('tag')
```

Comment:
Looks like `timestream-database` is supported in resourcegroupstaggingapi. We could potentially use that to read/write tags.
```
>>> response = client.tag_resources(ResourceARNList=['arn:aws:timestream:us-east-1:12356789123:database/sampleDB'], Tags={'c7n': 'test'})
>>> response = client.get_resources(ResourceARNList=['arn:aws:timestream:us-east-1:12356789123:database/sampleDB'])
>>> response
{'PaginationToken': '', 'ResourceTagMappingList': [{'ResourceARN': 'arn:aws:timestream:us-east-1:471176887411:database/sampleDB', 'Tags': [{'Key': 'Owner', 'Value': 'xyz'}, {'Key': 'c7n', 'Value': 'test'}]}]}
```

Reply:
weirdly enough i'm unable to tag via the resourcegroupstagging api in my account:

```
{'FailedResourcesMap': {'arn:aws:timestream:us-east-1:123456789012:database/test': {'StatusCode': 400, 'ErrorCode': 'InvalidParameterException', 'ErrorMessage': 'Unrecognized service or resource type for tagging'}}, 'ResponseMetadata': {'RequestId': '41827cc7-c48a-482e-9550-5cab0f87f569', 'HTTPStatusCode': 200, 'HTTPHeaders': {'x-amzn-requestid': '41827cc7-c48a-482e-9550-5cab0f87f569', 'content-type': 'application/x-amz-json-1.1', 'content-length': '208', 'date': 'Wed, 25 Jan 2023 22:34:16 GMT'}, 'RetryAttempts': 0}}
```

Reply:
you're right, I tested in some external accounts and it didn't work